src/track/steminfoimporter.cpp

```diff
@@ -0,0 +1,215 @@
+    auto file = QFile(filePath);
+    if (!file.open(QIODeviceBase::ReadOnly | QIODeviceBase::Unbuffered)) {
+        kLogger.warning()
+                << "Failed to open input file"
+                << filePath;
+        return {};
+    }
+
+    uint32_t manifestSize;
+    if (!(manifestSize = seekTillAtom(&file, kStemManifestAtomPath))) {
+        kLogger.debug()
+                << "No stem manifest found in the file"
+                << filePath;
+        return {};
+    }
+
+    auto jsonData = QJsonDocument::fromJson(file.read(manifestSize));
+    VERIFY_OR_DEBUG_ASSERT(jsonData.isObject()) {
+        kLogger.warning()
+                << "Failed to extract the manifest data"
+                << filePath;
+        return {};
+    };
+
+    auto json = jsonData.object();
+
+    // Check the manifest version
+    auto version = json.value("version").toInteger(-1);
+    if (version <= 0) {
+        kLogger.debug() << "Unexpected or missing version value in STEM manifest";
+        return {};
+    } else if (version > kSupportedStemVersion) {
+        kLogger.warning() << "Unsupported stem version" << version << "but trying anyway";
+    }
+
+    // Extract stem metadata
+    auto stems = json.value("stems");
+    if (!stems.isArray()) {
+        kLogger.debug() << "Unexpected or missing stems value in STEM manifest";
+        return {};
+    }
+    auto stemArray = stems.toArray();
+    QList<StemInfo> stemsList;
+    stemsList.reserve(stemArray.size());
+    int stemIdx = 0;
+    for (const auto& stemRef : stemArray) {
+        if (!stemRef.isObject()) {
+            kLogger.debug() << "Unexpected or missing stems value in STEM manifest";
+            return {};
+        }
+        auto stem = stemRef.toObject();
+        auto color = QColor(stem.value("color").toString());
+        auto name = stem.value("name").toString();
+        if (!color.isValid()) {
+            kLogger.debug() << "Unexpected or missing stem color in STEM manifest. Using default";
+            color = kStemDefaultColor[stemIdx];
+        }
+        if (name.isEmpty()) {
+            kLogger.debug() << "Unexpected or missing stem name in STEM manifest. Using default";
+            name = QObject::tr("Stem #%1").arg(QString::number(stemIdx + 1));
+        }
+        stemsList.emplace_back(name, color);
+        stemIdx++;
+    }
+
+    // Extract DSP information
+    // TODO(XXX) DSP only contains a limiter and a compressor effect, which
+    // Mixxx doesn't have yet. Parse and implement when supported
+    auto masteringDsp = json.value("mastering_dsp").toObject();
+    auto compressor = masteringDsp.value("compressor").toObject();
+    auto limiter = masteringDsp.value("limiter").toObject();
+
+    if (compressor.value("enabled").toBool() || limiter.value("enabled").toBool()) {
+        kLogger.debug() << "The STEM manifest is requesting DSP effects but "
+                           "this isn't supported by Mixxx yet.";
+    }
+
+    file.close();
```

Comment:
code style: isn't it a problem that this is not called on earlier exit paths of this func?

Reply:
[QFileDevice](https://doc.qt.io/qt-6/qfiledevice.html#dtor.QFileDevice) (which QFile inherit of) will ensure it closes it if necessary on destruction